### ISSUE TITLE
Windows Lens v5.0.0 first connection attempt fails

### ISSUE THREAD

**Issue by Kolossi** [lensapp/lens#3226]
**Describe the bug**
After adding external "sync" configs to lens for windows v5.0.0, first attempt to connect to the cluster always fails with "Invalid Credentials error"

**To Reproduce**
1. Add a cluster external config file as a "Kubeconfig Sync" in preferences "Kubernetes" tab
2. restart Lens
3. Attempt to connect to cluster 
4. Error "invalid credentials" is shown
5. Attempt to connect again and connection can suceed

**Expected behavior**
Where config is valid, first attempt should suceed without error.

**Screenshots**

Screenshot, including failed attempt to get debug logs for win version using WSL terminal:

![image](https://user-images.githubusercontent.com/5367728/123981046-eccd8080-d9b9-11eb-9f76-554a98634e2b.png)

**Environment (please complete the following information):**
- Lens Version: v5.0.0
- OS: Windows
- Installation method : auto upgraded from v5.0.0

**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:

Unable to get debug logs from windows lens - see later part of discussion on https://k8slens.slack.com/archives/C1U90NQN8/p1625045050141900 and screenshot in this post.

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.
```
N/A: connection works on second attempt
first time fail occurs with multiple clusters and with valid files using both ServiceAccount and Dex/LDAP login.
```
** Additional **

Would like to grab and post debug logging, will do so when advised how to do it in windows version of lens.

**Comment by jeffjones-kbx** [comment]
I am also getting the failed connect at initial connection to a cluster. Hit Reconnect a few times and got further but received the output in the screenshot below.  I am able to view the cluster resources now though.

![image](https://user-images.githubusercontent.com/77127429/123982229-76645a80-d988-11eb-9a30-c5fe580820c6.png)

**Comment by Kolossi** [comment]
Additional observation:

When performing the initial failing connect attempt there is only the message "Connecting..." and no message "Authentication proxy started" at any stage.

When clicking to try again there is the message "Reconnecting..." and then after a while "Authentication proxy started" after which the connection then succeeds.

**Comment by Nokel81** [comment]
That is a good hint thanks

**Comment by andyliddle** [comment]
Can take 2 or 3 times to connect and it's not quick compared to previous versions.

Also, I just get the error "Oops, something went wrong."

**Comment by kumargauravsinha** [comment]
When can we expect the fix for this recurring issue please ? This is so frustrating

**Comment by fatpowaranga** [comment]
I assumed this would be resolved before 5.0 release since it was noted by the team in the beta, but it's still a very prevalent issue. Now when Lens has very high CPU usage (after opening a couple clusters or so now) and I go to restart Lens, I run into this error for every cluster individually and it can take 1 to 5 reconnection attempts before it eventually works.

Any thoughts on what could be the cause? 

Basically: 

1. Try to connect to cluster after restarting Lens.
2. Get "Authentication proxy started - Oops, something went wrong" and hit Reconnect
3. Most of the time it connects here, sometimes it goes into a bit of a loop and then eventually reconnects after a few attempts.

I have noticed that if I then immediately or in the next few minutes try to connect to another cluster, it works more often than not.

**Comment by Nokel81** [comment]
Marking this as a blocker for 5.1.1 as we plan to release 5.1 (with some other bug fixes) today.

**Comment by maartengo** [comment]
Updated just now, the first connection fails with the following error: `Oops, something went wrong. Error: connect ECONNREFUSED 127.0.0.1:80`
![image](https://user-images.githubusercontent.com/46345801/125904639-5ed6a994-fe22-496c-91b0-16bcf85dd8d7.png)

This is on 5.1.0-latest.20210715.1

**Comment by ramanNarasimhan77** [comment]
I can also confirm that this is still happening in the latest release build

![image](https://user-images.githubusercontent.com/52691361/125905612-72568b19-d293-4c90-9794-fc9e2e181647.png)





![image](https://user-images.githubusercontent.com/52691361/125905659-de59c15c-812c-4120-9a32-0724a21086b6.png)

**Comment by Nokel81** [comment]
Yes we are currently investigating why that proxy connection is failing on windows. Thanks for the additional report.

**Comment by m8ram** [comment]
Every first connection to a cluster fails. Example for an AWS EKS cluster:
![image](https://user-images.githubusercontent.com/6831686/126133490-2a5f3572-fc80-4128-b9fd-472561e946c4.png)

This fails in the same way for docker desktop kubernetes cluster, GCP K8s cluster.

I have my K8s kubeconfig set up locally, no sync or Lens cloud.

After clicking reconnect it always worked so far.

Is the local web server (127.0.0.1:80) part of Lens?

**Comment by Nokel81** [comment]
Yes the local web server is part of Lens, we are still looking into this.

**Comment by maartengo** [comment]
https://github.com/lensapp/lens/pull/3458 seems to have fixed the issue 🎉
![image](https://user-images.githubusercontent.com/46345801/126758807-8d1b4224-0bcd-4238-be2f-06b062fb8ffb.png)

**Comment by Nokel81** [comment]
🎉 Excellent, what about you @Kolossi ?

**Comment by ramanNarasimhan77** [comment]
I also confirm that the latest build has fixed this issue.

![image](https://user-images.githubusercontent.com/52691361/126821524-f57f48c2-a34b-4bb6-870d-51f997db16e2.png)

**Comment by Kolossi** [comment]
@Nokel81 : Unfortunately not quite there for me on Windows 10 and exactly same "help->about" as image in ramanNarasimhan77 previous post.

When I click to connect to first cluster, it goes through "connecting..." ->5secs delay -> "Authentication proxy started", but then spins forever (at least to 5 mins), no error shown.

If I click a second cluster I again get "connecting..." -> "Authentication proxy started", but then spins forever, no error shown.

Clicking back on the first cluster, it says "connecting..." then connection completes after about a second and overview shown.

Clicking back on the second cluster, it says "connecting..." then connection completes after about a second and overview shown.

Having done this, if I click on a third cluster I still get "connecting..." -> "Authentication proxy started" and it spins forever, no error shown.

Clicking off to first or second cluster then back to third cluster, it says "connecting..." then connection completes after about a second and overview shown.

**Comment by HarelM** [comment]
I'm too experiencing connection issues with Lens 5. It doesn't get solved for me in the second attempt. Not sire if I need to open a separate issue for this.
I have downgraded to version 4.2.5 and connection is working.
Let me know if I can help in any way - logs etc.
kubectl is working fine without any issues using the same config file.

**Comment by Kolossi** [comment]
@Nokel81 Any luck tracking this down?

Mostly it still does as per [my previous comment](https://github.com/lensapp/lens/issues/3226#issuecomment-887362311), but just occasionally (like this morning) it connects straight in.

This does make me wonder again (with no evidence other than suspicion) whether https://github.com/lensapp/lens/issues/2750 is involved?

**Comment by Nokel81** [comment]
We have https://github.com/lensapp/lens/pull/3511 which is slated to be part of the 5.2.0 release. As for 2750, I wonder if that is a race condition... We use the port that is automatically given to us by the OS. However, in 4.2.4 there was a race condition where we would try and get a port and then stop using it before trying to use it again, that has been removed.

**Comment by m8ram** [comment]
I still experienced the issue with 5.1.3 as well.
With 5.2.0 on first use the connections did not throw these errors. I'll monitor in the coming days.

**Comment by Kolossi** [comment]
@Nokel81 I agree with @m8ram that initial indications with 5.2.0 are good - thanks!

As it's an issue that sometimes happens and sometimes doesn't, it seems fair to hold on closing for now though - I'll also monitor over the next few days and report back.

**Comment by Nokel81** [comment]
Will close this then. Please open a new issue (referencing this one) if you encounter it again.

**Comment by HarelM** [comment]
It's now connecting in version 5.2 but causes a blue screen of death in Windows... I'll open a new issue.
